src/devices_models/devices/common/add_variable.jl

```diff
@@ -19,7 +19,11 @@ function add_variables!(
     service::U,
     contributing_devices::Union{Vector{V}, IS.FlattenIteratorWrapper{V}},
     formulation::AbstractReservesFormulation,
-) where {T <: VariableType, U <: PSY.Reserve, V <: PSY.Component}
+) where {
+    T <: VariableType,
+    U <: Union{PSY.Reserve, PSY.ReserveNonSpinning},
```

Comment:
This shouldn't be necessary. Isn't `PSY.ReserveNonSpinning <: PSY.Reserve`

Reply:
Nope, both are sub types of `<: Service`

Reply:
Then we need to fix PSY first